iceberg/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuSparkBatchQueryScan.java

```diff
@@ -139,12 +139,8 @@ private static TableScan buildScan(Scan cpuInstance,
       scan = scan.asOfTime(asOfTimestamp);
     }
 
-    if (startSnapshotId != null) {
-      if (endSnapshotId != null) {
-        scan = scan.appendsBetween(startSnapshotId, endSnapshotId);
-      } else {
-        scan = scan.appendsAfter(startSnapshotId);
-      }
+    if (startSnapshotId != null || endSnapshotId != null) {
+      throw new UnsupportedOperationException("Incremental scan is not supported");
```

Comment:
So is this the long term plan? If so how do we fall back to the CPU when this functionality is needed? If not please make sure we have a P0 blocking issue filed to make sure we don't forget this for the release.